From 3b32931252a51d5aed91e2028928be907ebdff7c Mon Sep 17 00:00:00 2001
From: Matthew Hui <mhui@codeaurora.org>
Date: Sun, 1 Jan 2012 16:51:26 -0500
Subject: [PATCH] external/webkit: use Cpu Upload path

Use Cpu Upload mode since the Gpu mode is not working
and expected to be slower.

Change-Id: Iacb6de4a4e054e4f6ebb07d8a8209228e78e5ee0
---
 Source/WebCore/Android.mk                                            | 5 +++++
 Source/WebCore/platform/graphics/android/rendering/TransferQueue.cpp | 4 ++++
 Source/WebCore/platform/graphics/android/rendering/TransferQueue.h   | 4 ++++
 3 files changed, 13 insertions(+)

diff --git a/Source/WebCore/Android.mk b/Source/WebCore/Android.mk
index b1b430c..1f8868e 100644
--- a/Source/WebCore/Android.mk
+++ b/Source/WebCore/Android.mk
@@ -1310,3 +1310,8 @@ ifeq ($(SUPPORT_COMPLEX_SCRIPTS),true)
 LOCAL_SRC_FILES := $(LOCAL_SRC_FILES) \
 	platform/graphics/android/fonts/HarfbuzzSkia.cpp
 endif
+
+# HACK for qcom devices CPU upload mode
+ifeq ($(TARGET_FORCE_CPU_UPLOAD),true)
+LOCAL_CFLAGS += -DFORCE_CPU_UPLOAD
+endif
diff --git a/Source/WebCore/platform/graphics/android/rendering/TransferQueue.cpp b/Source/WebCore/platform/graphics/android/rendering/TransferQueue.cpp
index 7c03219..857d377 100644
--- a/Source/WebCore/platform/graphics/android/rendering/TransferQueue.cpp
+++ b/Source/WebCore/platform/graphics/android/rendering/TransferQueue.cpp
@@ -516,7 +516,11 @@ void TransferQueue::setTextureUploadType(TextureUploadType type)
 
     setPendingDiscard();
 
+#ifdef FORCE_CPU_UPLOAD
+    m_currentUploadType = CpuUpload; // force to cpu upload mode for now until gpu upload mode is fixed
+#else
     m_currentUploadType = type;
+#endif
     ALOGD("Now we set the upload to %s", m_currentUploadType == GpuUpload ? "GpuUpload" : "CpuUpload");
 }
 
diff --git a/Source/WebCore/platform/graphics/android/rendering/TransferQueue.h b/Source/WebCore/platform/graphics/android/rendering/TransferQueue.h
index 55011b0..4ced1ea 100644
--- a/Source/WebCore/platform/graphics/android/rendering/TransferQueue.h
+++ b/Source/WebCore/platform/graphics/android/rendering/TransferQueue.h
@@ -69,7 +69,11 @@ enum TextureUploadType {
     GpuUpload = 1
 };
 
+#ifdef FORCE_CPU_UPLOAD
+#define DEFAULT_UPLOAD_TYPE CpuUpload
+#else
 #define DEFAULT_UPLOAD_TYPE GpuUpload
+#endif
 
 class TileTransferData {
 public:
-- 
1.8.4

